internal/cmd/skupper/token/nonkube/token_redeem.go

```diff
@@ -1,30 +1,133 @@
 package nonkube
 
 import (
+	"errors"
 	"fmt"
+	"github.com/skupperproject/skupper/internal/nonkube/client/fs"
+	nonkubecommon "github.com/skupperproject/skupper/internal/nonkube/common"
+	"github.com/skupperproject/skupper/internal/utils/validator"
+	"github.com/skupperproject/skupper/pkg/apis/skupper/v2alpha1"
+	"github.com/skupperproject/skupper/pkg/nonkube/api"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"os"
 
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
+	"github.com/skupperproject/skupper/pkg/generated/client/clientset/versioned/scheme"
 	"github.com/spf13/cobra"
 )
 
 type CmdTokenRedeem struct {
-	CobraCmd  *cobra.Command
-	Flags     *common.CommandTokenRedeemFlags
-	Namespace string
-	siteName  string
+	CobraCmd      *cobra.Command
+	Flags         *common.CommandTokenRedeemFlags
+	Namespace     string
+	siteName      string
+	linkHandler   *fs.LinkHandler
+	secretHandler *fs.SecretHandler
+	fileName      string
+	name          string
+	siteState     *api.SiteState
 }
 
 func NewCmdTokenRedeem() *CmdTokenRedeem {
 	return &CmdTokenRedeem{}
 }
 
 func (cmd *CmdTokenRedeem) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.Namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.linkHandler = fs.NewLinkHandler(cmd.Namespace)
+	cmd.secretHandler = fs.NewSecretHandler(cmd.Namespace)
 }
 
-func (cmd *CmdTokenRedeem) ValidateInput(args []string) error { return nil }
-func (cmd *CmdTokenRedeem) InputToOptions()                   {}
+func (cmd *CmdTokenRedeem) ValidateInput(args []string) error {
+	var validationErrors []error
+	tokenStringValidator := validator.NewFilePathStringValidator()
+
+	// Validate token file name
+	if len(args) < 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("token file name must be configured"))
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("file name must not be empty"))
+	} else {
+		ok, err := tokenStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("token file name is not valid: %s", err))
+		} else {
+			cmd.fileName = args[0]
+		}
+	}
+
+	// Validate there is already a site defined in the namespace before a token can be redeemed
+	pathProvider := fs.PathProvider{Namespace: cmd.Namespace}
+	siteStateLoader := &nonkubecommon.FileSystemSiteStateLoader{
+		Path: pathProvider.GetRuntimeNamespace(),
+	}
+
+	siteState, err := siteStateLoader.Load()
```

Comment:
I believe this could be avoided. The reason, as I can see, for a site to exist is just to provide a site name
as the subject when redeeming the AccessToken. But I believe that, instead, if we could pass the namespace
name as an alternative, we should be good, then there is no need to require an active site to exist in advance.